packages/core/lib/process-request.ts

```diff
@@ -263,25 +264,39 @@ export const processRequest = async <TContext = {}, TRootValue = {}>(
           // Operations that use @defer, @stream and @live will return an `AsyncIterable` instead of an
           // execution result.
           if (isAsyncIterable<ExecutionPatchResult>(result)) {
-            return {
-              type: isHttpMethod("GET", request.method) ? "PUSH" : "MULTIPART_RESPONSE",
-              subscribe: async (onResult) => {
-                for await (const payload of result) {
-                  onResult(
-                    formatPayload({
-                      payload,
-                      context,
-                      rootValue,
-                      document,
-                      operation,
-                    })
-                  );
+            const asyncSubscribe = async (onResult: Function) => {
```

Comment:
Is there any way to improve how we could type it? I wouldn't like to type it as `Function`, seems very weird.

I was thinking about an union like: `MultipartResponseSubscribe | PushSubscribe`, what do you think?

Reply:
That sounds good!

Reply:
> That sounds good!

@n1ru4l, I did some improvements on it, but I wasn't like 100% of it. Any feedback about how we could improve it?